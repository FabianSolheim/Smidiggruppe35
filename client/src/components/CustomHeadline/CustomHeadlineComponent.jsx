import React from "react";
import "./CustomHeadlineStyle.css";
import { Typography } from "@mui/material";

const CustomHeadline = ({ size, children, ...otherProps }) => {
  //CustomHeadline takes in a size prop and a children prop.
  //The size prop is used to determine the size of the headline.
  //The children prop is used to display the text.

  return (
    <div className="headline-container">
      <Typography variant={size} className="title" {...otherProps}>
        {children}
      </Typography>
      <div className="line" />
    </div>
  );
};

export default CustomHeadline;
