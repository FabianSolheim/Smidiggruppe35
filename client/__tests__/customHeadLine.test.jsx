import React from "react";
import ReactDOM from "react-dom";
import CustomHeadlineComponent from "../src/components/CustomHeadline/CustomHeadlineComponent";

describe("customHeadlineComponent", () => {
  it("shows customHeadlineComponent", () => {
    const domElement = document.createElement("div");
    ReactDOM.render(<CustomHeadlineComponent />, domElement);
    expect(domElement.innerHTML).toMatchSnapshot();
  });
});
