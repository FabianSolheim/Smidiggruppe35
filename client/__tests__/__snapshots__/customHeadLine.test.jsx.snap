// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customHeadlineComponent shows customHeadlineComponent 1`] = `"<div class=\\"headline-container\\"><p class=\\"MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root\\"></p><div class=\\"line\\"></div></div>"`;
